### ISSUE TITLE
Questions about the paper

### ISSUE THREAD

**Issue by OMEGA-Y** [ArmastusChen/Learning-Continuous-Implicit-Representation-for-Near-Periodic-Patterns#2]
Hi, thanks for your great work.
I have two questions about reading the paper and the code in the image completion task.

Q. Do you mask the image randomly and input the NPPnet?
Q. What is the goal of training NPPnet? I guess we train the MLP through the pixel and patch loss, and then we get the MLP that has an RGB value for each coordinate in the image. Can we get the periodicity or the unit patch in the repeated pattern as well as such MLP?

**Comment by ArmastusChen** [comment]
Hi,

Thanks for the interest to our work.

Q: Do you mask the image randomly and input the NPPnet?

A: For details of mask generation, please refer to the section 3.1 in the supplementary of our arXiv version (https://arxiv.org/pdf/2208.12278.pdf).

Q: What is the goal of training NPPnet? I guess we train the MLP through the pixel and patch loss, and then we get the MLP that has an RGB value for each coordinate in the image. 

A: The goal of training NPPNet is to learn the representation of a specific NPP image. After training, you can do completion, segmentation, or deblurring as we discussed in the paper. 

Q: Can we get the periodicity or the unit patch in the repeated pattern as well as such MLP?

A: We have already detected the periodicity before we train the MLP.  The learned MLP is for the whole image but you can sample a unit patch from the MLP based on the periodicity.

Please let me know if you have more questions.

**Comment by OMEGA-Y** [comment]
Thanks for your answer!
I am still confused about the overall process. So, can you check my understanding?

For training, suppose that we can only observe a partial part of an image since we mask the input and use that part for the ground truth. (Is it self-supervised learning and proper the reason?)
NPPnet first predicts the periodicity of the image. And then, warped coordinates and original coordinates in the image are fed into the MLP so that MLP learns more precise RGB values for each coordinate in the image.
Finally, we can do completion, segmentation, or deblurring using the learned MLP. (Do we need some fine-tuning and additional layers for each downstream task? And do we measure the evaluation metric on the test set by MLP optimized with a single image?)

Thank you very much for taking the time to read it.

**Comment by ArmastusChen** [comment]
In the main paper, we use the completion task as example to demonstrate NPPNet. So the pipeline figures in the main paper are specifically for the main paper. NPP-Net can be adapted to the other two applications with minor modification. We only assume partial observation for completion task, and assume the full image is available in segmentation and deblurring tasks. You can also check the "./data" folder in this repo for understanding.

Q: Is it self-supervised learning and proper the reason?

Yes, for completion, we supervised NPPNet in the known region. For the other two tasks, check the supplementary for more details.

Q: Do we need some fine-tuning and additional layers for each downstream task?

No, we don't need fine-tuning and additional layers. The NPP-Net is trained from scratch for each task. In other words, even for the same image, we are not using the same learned MLP for the three tasks. 

Q:  Do we measure the evaluation metric on the test set by MLP optimized with a single image?

For completion task, the test set for a single image is the unknown region (we know the GT because we generate the mask on the image). See supplementary for the other two tasks.



Please let me know if you have more questions.

**Comment by OMEGA-Y** [comment]
Thanks to your detailed answer, I understood the overall process in this paper😄

I will close this issue.
